server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -68,14 +70,23 @@ public class ClusterStatsIndices implements ToXContentFragment {
     private MappingStats mappings;
 
     public ClusterStatsIndices(List<ClusterStatsNodeResponse> nodeResponses, MappingStats mappingStats, AnalysisStats analysisStats) {
-        Map<String, ShardStats> countsPerIndex = new HashMap<>();
+        this(Set.of(IndexMetric.values()), nodeResponses, mappingStats, analysisStats);
+
+    }
 
-        this.docs = new DocsStats();
-        this.store = new StoreStats();
-        this.fieldData = new FieldDataStats();
-        this.queryCache = new QueryCacheStats();
-        this.completion = new CompletionStats();
-        this.segments = new SegmentsStats();
+    public ClusterStatsIndices(
```

Comment:
Since we deleted the old constructor, we may get backward compatibility check failures in 2.x.
Could you verify the same and ensure we handle it?

Reply:
Old constructor is still there for this class. Change has been made to call the new constructor from the old constructor with default arguments.